### ISSUE TITLE
Behave different with quite nvim activate virtual env and enter the nvim!

### ISSUE THREAD

**Issue by meicale** [AckslD/swenv.nvim#28]
Behave differently with quite nvim activate virtual env and enter the nvim!
I used this plugin to change used virtual env on the fly when run shell command directly in neovim. It works and it can find the right python bin. But the editor can not notice that changes. The diagnostic tools still complains the wrong things. Does anyone know how to solve this? Thanks!

**Comment by AckslD** [comment]
Hi @meicale, probably dependent on how you configure your diagnostic tools?

**Comment by Klvxn** [comment]
Any update on this? I've observed the same thing.

**Comment by AckslD** [comment]
I would need some more details to be able to help here, it does work for me

**Comment by Klvxn** [comment]
- when venv is activated before launching neovim. 
 "Go to Declaration" on a class or function from third party package, takes you to package in the activated virtual env 

- Activating with swenv.
"Go to Declaration" takes you to the system wide installed package. 


And I noticed in your `update_path` function, the PATH variable gets updated with `~/project/venv/bin` which works as expected for Linux/Mac. I run Windows and `.../venv/bin` doesn't exist. `.../venv/scripts` is for Windows

**Comment by AckslD** [comment]
I see, I don't know how venvs work on windows. On UNIX systems it's sufficient to update PATH. What would be the corresponding action on windows?

**Comment by Klvxn** [comment]
Yeah, updating PATH is sufficient but your implementation updates the wrong venv_path for Windows. 

I manually changed the `update_path` function to this and it works as it should. 

```
local update_path = function(path)
    vim.fn.setenv('PATH', path .. '/Scripts' .. ';' .. ORIGINAL_PATH)
    -- /Scripts instead of /bin
end
```

The lsp now opens packages installed in the activated virtual environment. 

It's like I said before, you initially used `venv_path/bin` to update the PATH variable. But on Windows, it's `venv_path/Scripts`
The Scripts dir is where the python executable is located.

**Comment by AckslD** [comment]
Thanks @Klvxn! Would you like to open a PR which fixes this with your `update_path` if you're on windows?

**Comment by Klvxn** [comment]
Okay I just went through previous PRs and someone already opened one for the same exact purpose (#43 ). So, I'd simply just leave a review of the changes the person made if that's okay?

**Comment by AckslD** [comment]
Oh right thanks for pointing that out, it had completely slipped my mind
